src/RBTree.mo

```diff
@@ -209,8 +209,11 @@ module {
   public func size<X, Y>(t : Tree<X, Y>) : Nat {
     switch t {
       case (#leaf) { 0 };
-      case (#node(_, l, _, r)) {
-        size(l) + size(r) + 1
+      case (#node(_, l, xy, r)) {
+        switch (xy.1) {
+          case null { size(l) + size(r) };
+          case (_) { size(l) + size(r) + 1 };
+        }
```

Comment:
```suggestion
      case (#node(_, l, xy, r)) {
        size(l) + size(r) + switch (xy.1) { case null 0; case _ 1 }
      }
```

Reply:
I considered that before making this PR, but I decided that the current code as is is slightly easier to parse and understand for a new reader.

Reply:
As a compiler engineer I often think of reducing Wasm size (which comes at a premium), and since `moc` can't perform this optimisation yet, I tend to do it manually.

Reply:
Done.